web/services/premium_detection.go

```diff
@@ -0,0 +1,65 @@
+package services
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/trento/version"
+)
+
+const Premium = "Premium"
+
+type PremiumDetection interface {
+	ShouldRequireEulaAcceptance() (bool, error)
+	CanSendTelemetry() bool
+	SupportsPremiumChecks() bool
+}
+
+type premiumDetection struct {
+	subscriptions SubscriptionsService
+	settings      SettingsService
+}
+
+func (premiumDetection *premiumDetection) ShouldRequireEulaAcceptance() (bool, error) {
+	if !premiumDetection.isPremiumFlavor() {
+		return false, nil
+	}
+
+	isEulaAccepted, err := premiumDetection.settings.IsEulaAccepted()
+	if err != nil {
+		log.Warnf("Unable to determine whether the EULA for the Trento installation has been accepted. Assuming it is not. Error: %s", err)
+	}
+	return !isEulaAccepted, err
+}
+
+func (premiumDetection *premiumDetection) CanSendTelemetry() bool {
+	if !premiumDetection.isPremiumFlavor() {
+		return false
+	}
+
+	isEulaAccepted, err := premiumDetection.settings.IsEulaAccepted()
+	if err != nil {
+		log.Warnf("Unable to determine whether the EULA has been accepted for telemetry. Assuming it is not. Error: %s", err)
+	}
+	return isEulaAccepted
+}
+
+func (premiumDetection *premiumDetection) SupportsPremiumChecks() bool {
+	if !premiumDetection.isPremiumFlavor() {
+		return false
+	}
+	isPremiumActive, err := premiumDetection.subscriptions.IsTrentoPremium()
+	if err != nil {
+		log.Warnf("Unable to determine whether the Trento Premium installation is active. Assuming it is not. Error: %s", err)
+	}
+	return isPremiumActive
+}
+
+func (premiumDetection *premiumDetection) isPremiumFlavor() bool {
+	return Premium == version.Flavor
+}
+
+func NewPremiumDetection(subscriptions SubscriptionsService, settings SettingsService) PremiumDetection {
```

Comment:
This should return better a `*premiumDetection`.
And I would put it on the top of the file

Reply:
changed